src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -45,20 +47,56 @@ public static string HtmlEncode(string value)
                 return value;
             }
 
-            StringBuilder sb = StringBuilderCache.Acquire(value.Length);
-            HtmlEncode(value, index, sb);
-            return StringBuilderCache.GetStringAndRelease(sb);
+            var sb = new PooledStringBuilder(value.Length);
+            try
```

Comment:
You do not need try/catch. It is ok to turn the pooled into garbage on exceptions.